### PR TITLE
Feat: Podio OAuth provider

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

OAuth2 implementation with [Podio](https://podio.com/). Podio (as far as I am concerned) is a database and a CMS, both in one product.

## Test Plan

- [x] Manual QA:

image

image

image

CleanShot 2022-07-02 at 14 26 22

## Related PRs and Issues

None, a friend of mine asked for it 😆

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes 😎

### CODE REVIEW

**Comment by MatusFercak** [comment]
GG

File: src/Appwrite/Auth/OAuth2/Podio.php [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                'POST',
+                $this->apiEndpoint . '/oauth/token',
+                ['Content-Type: application/x-www-form-urlencoded'],
+                \http_build_query([
+                    'grant_type' => 'authorization_code',
+                    'code' => $code,
+                    'redirect_uri' => $this->callback,
+                    'client_id' => $this->appID,
+                    'client_secret' => $this->appSecret,
+                    // 'scope' => \implode(' ', $this->getScopes())
```
Comment (christyjacob4) [inline src/Appwrite/Auth/OAuth2/Podio.php:84]:
```suggestion
```

File: src/Appwrite/Auth/OAuth2/Podio.php [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserName(string $accessToken): string
+    {
+        return '';
```
Comment (christyjacob4) [inline src/Appwrite/Auth/OAuth2/Podio.php:171]: Does the API not provide the name of the user.?

Comment (christyjacob4) [reply src/Appwrite/Auth/OAuth2/Podio.php:171]: Is there another endpoint that can be used? Like a `/profile` endpoint ?

**Review by christyjacob4 — CHANGES_REQUESTED** [review]

File: src/Appwrite/Auth/OAuth2/Podio.php (hunk 2 above)
Comment (Meldiron) [reply src/Appwrite/Auth/OAuth2/Podio.php]: Nice idea! Yep, possible. Implemented.

CleanShot 2022-07-04 at 12 32 36

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
Nice one! 👌 👀

**Comment by christyjacob4** [comment]
@Meldiron  can you please sync this with master once again?

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**